web/app/index.js

```diff
@@ -8,7 +8,7 @@ function onNavigate (evt) {
   }
   const args = hash.substring(1, hash.length).split(',');
   const name = args[0];
-  const ele = document.querySelector(`[args="${hash}"]`) || document.createElement(name);
+  const ele = document.querySelector(`[identifier="${name}"]`) || document.createElement(name);
```

Comment:
this breaks once there are multiple panels with the same element

Reply:
Replaces panels with the same element otherwise we would have 3 panels for each of the actions. Not sure there is a better approach?

Reply:
We have `ele.remove()` just below this line which ensures only one panel of same element?

Reply:
We append it again in the DOM. This keeps the element positions 'sorted'.

Reply:
Looks like `postMessage` is way to go given the problem of multiple habitat-account panels 🙂 .
For example, habitat-vault,<id> can have multiple panels. And special casing habitat-account is odd too.

Reply:
Yes but the fact that we remove before appending it ensures single element. 
- `#habitat-account,Withdraw` 
- `#habitat-account,Deposit`
- `#habitat-account`
Given these 3 routes we will have one element of type `habitat-account`  with current implementation. No duplicates.
Unless we have a scenario where we want to load two or more panels of `habitat-account` or any other type? I can change to use postMessage if that's the case

Reply:
let's say we have:

`habitat-vault,id2`
`habitat-vault,id1`

this would be broken now, because `identifier="habitat-vault"` can indeed have multiple `unique` elements in page. `args="habitat-vault,id1`

Reply:
Used `window.postMessage` can be extracted to utils the more same structure is used to update components?

Reply:
> Used `window.postMessage` can be extracted to utils the more same structure is used to update components?

We can defer that until we use that in more places 👍